api/server/server.go

```diff
@@ -130,6 +130,9 @@ func parseMultipartForm(r *http.Request) error {
 }
 
 func httpError(w http.ResponseWriter, err error) {
+	if err == nil || w == nil {
+		return
```

Comment:
@WeiZhang555 I doubt we ever call httpError with a nil `w` or a nil `err`, so I think we should add a log message before returning. `logrus.WithFields(logrus.Fields{"error": err, "writer": w}).Error("unexpected")`